presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/GroupTrait.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sql.planner.iterative;
+
+import static java.util.Objects.requireNonNull;
+
+public interface GroupTrait
```

Comment:
It's not obvious why we need a separation between `GroupTrait` and just a `Trait`. After all, one can say that any node (and not group as whole) within group has the same cardinality or stats. I thing `Lookup` can easily mask if a trait is computed per node or per group. Creating such split early on seems unnecessary and will require different code paths for group/non-group traits.

I think the fact that a trait can be computed once per group is rather an optimization, so we should figure a way of telling to the memo, that: hey, we have computed a trait for a plan node, but you can as well apply it for all nodes within group.